File: server/models/company.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

// type CompanyBasic struct{
// 	ObjectId primitive.ObjectID `bson:"_id"`
// 	Id int32 `bson:"id"`
// 	Short string
// 	CompanyName string
// 	City string
// 	Services []string
// 	Status string
// }

type General struct {
    CompanyName           string    `bson:"CompanyName"`
    CompanyShort          string    `bson:"CompanyShort"`
    CompanyId            int       `bson:"CompanyId"`
    Services              []string  `bson:"Services"`
    ESIPFCreatedBy        string    `bson:"ESIPFCreatedBy"`
    CompanyType           string    `bson:"CompanyType"`
    Status                string    `bson:"Status"`
    BusinessAddress       string    `bson:"BusinessAddress"`
    City                  string    `bson:"City"`
    PostalCode            string    `bson:"PostalCode"`
    District              string    `bson:"District"`
    State                 string    `bson:"State"`
    ContactPersonName     string    `bson:"ContactPersonName"`
    ContactPersonRole     string    `bson:"ContactPersonRole"`
    ContactPersonMobile   string    `bson:"ContactPersonMobile"`
    ContactPersonWhatsapp string    `bson:"ContactPersonWhatsapp"`
    ContactPersonMailId   string    `bson:"ContactPersonMailId"`
}

type Partner struct {
    AadhaarName    string    `bson:"AadhaarName"`
    DateOfBirth    time.Time `bson:"DateOfBirth"`
    AadhaarNumber  string    `bson:"AadhaarNumber"`
    Address        string    `bson:"Address"`
    PAN            string    `bson:"PAN"`
    PANName        string    `bson:"PANName"`
    Mobile         string    `bson:"Mobile"`
    Email          string    `bson:"Email"`
    Photo          string    `bson:"Photo"`
}

type CompanyDetails struct {
    CompanyPAN         string    `bson:"CompanyPAN"`
    PANDate            time.Time `bson:"PANDate"`
    PANName            string    `bson:"PANName"`
    GSTDate            time.Time `bson:"GSTDate"`
    BankAccountName    string    `bson:"BankAccountName"`
    BankAccountNumber  string    `bson:"BankAccountNumber"`
    IFSC               string    `bson:"IFSC"`
    BankType           string    `bson:"BankType"`
    BankName           string    `bson:"BankName"`
    BranchName         string    `bson:"BranchName"`
    LIN                string    `bson:"LIN"`
    CIN                string    `bson:"CIN"`
    CINDate            time.Time   `bson:"CINDate"`
    BranchAddress      string    `bson:"BranchAddress"`
    NatureOfBusiness   string    `bson:"NatureOfBusiness"`
    GSTNumber          string    `bson:"GSTNumber"`
    City               string    `bson:"City"`
    PostalCode         string    `bson:"PostalCode"`
    District           string    `bson:"District"`
    State              string    `bson:"State"`
    PartnerList        []Partner `bson:"PartnerList"`
}

type EPFO struct {
    EPFOStatus            string    `bson:"EPFOStatus"`
    EPFOLogin             string    `bson:"EPFOLogin"`
    EPFOPassword          string    `bson:"EPFOPassword"`
    EPFOCode              string    `bson:"EPFOCode"`
    EPFOCodeType          string    `bson:"EPFOCodeType"`
    EPFOMailId            string    `bson:"EPFOMailId"`
    EPFOMobile            string    `bson:"EPFOMobile"`
    EPFOFilingSince       time.Time `bson:"EPFOFilingSince"`
    EPFODateOfCoverage    time.Time `bson:"EPFODateOfCoverage"`
    EPFOCoverageType      string    `bson:"EPFOCoverageType"`
    DSCStatus             string    `bson:"DSCStatus"`
    DSCExpiryDate         time.Time `bson:"DSCExpiryDate"`
    DSCPassword           string    `bson:"DSCPassword"`
    ESignStatusEPFO       string    `bson:"ESignStatusEPFO"`
    EPFOOfficeName        string    `bson:"EPFOOfficeName"`
    EPFOOfficeAddress     string    `bson:"EPFOOfficeAddress"`
    EPFOOfficeContactNumber string  `bson:"EPFOOfficeContactNumber"`
}

type ESIC struct {
    ESICStatus             string    `bson:"ESICStatus"`
    ESICLogin              string    `bson:"ESICLogin"`
    ESICPassword           string    `bson:"ESICPassword"`
    ESICCode               string    `bson:"ESICCode"`
    ESICCodeType           string    `bson:"ESICCodeType"`
    ESICMailId             string    `bson:"ESICMailId"`
    ESICMobile             string    `bson:"ESICMobile"`
    ESICFilingSince        time.Time `bson:"ESICFilingSince"`
    ESICDateOfCoverage     time.Time `bson:"ESICDateOfCoverage"`
    ESICOfficeName         string    `bson:"ESICOfficeName"`
    ESICOfficeAddress      string    `bson:"ESICOfficeAddress"`
    ESICOfficeContactNumber string  `bson:"ESICOfficeContactNumber"`
}

type ShramSuvidha struct {
    ShramSuvidhaLogin    string    `bson:"ShramSuvidhaLogin"`
    ShramSuvidhaPassword string    `bson:"ShramSuvidhaPassword"`
    ShramSuvidhaMailId   string    `bson:"ShramSuvidhaMailId"`
    ShramSuvidhaMobile   string    `bson:"ShramSuvidhaMobile"`
    NearestPoliceStation string    `bson:"NearestPoliceStation"`
    NICCode              string    `bson:"NICCode"`
    BusinessPlace        string    `bson:"BusinessPlace"`
    DateOfApplication    time.Time `bson:"DateOfApplication"`
    LeaseDateFrom        time.Time `bson:"LeaseDateFrom"`
    LeaseDateTo          time.Time `bson:"LeaseDateTo"`
}

type IF struct {
    IFStatus             string    `bson:"IFStatus"`
    IFLogin              string    `bson:"IFLogin"`
    IFPassword           string    `bson:"IFPassword"`
    IFLicenseNumber      string    `bson:"IFLicenseNumber"`
    IFMailId             string    `bson:"IFMailId"`
    IFMobile             string    `bson:"IFMobile"`
    IFDateOfRegistration time.Time `bson:"IFDateOfRegistration"`
    IFRenewalCharges     int       `bson:"IFRenewalCharges"`
    IFRenwalSince        time.Time `bson:"IFRenwalSince"`
    NumberOfPressureVessels int     `bson:"NumberOfPressureVessels"`
    ISMLogin             string    `bson:"ISMLogin"`
    ISMPassword          string    `bson:"ISMPassword"`
}

type Contact struct {
    ContactName   string `bson:"ContactName"`
    Designation   string `bson:"Designation"`
    Email         string `bson:"Email"`
    Mobile        string `bson:"Mobile"`
}

type Fee struct {
    Services []string `bson:"Services"`
    Period   string   `bson:"Period"`
    Fee      int      `bson:"Fee"`
}

type Registration struct {
    EPFO       EPFO       `bson:"EPFO"`
    ESIC       ESIC       `bson:"ESIC"`
    ShramSuvidha ShramSuvidha `bson:"ShramSuvidha"`
    IF         IF         `bson:"IF"`
}

type Company struct {
    Id           primitive.ObjectID `bson:"_id,omitempty"`
    General      General             `bson:"General"`
    CompanyDetails CompanyDetails    `bson:"CompanyDetails"`
    Registration   Registration       `bson:"Registration"`
    ContactList []Contact `bson:"ContactList"`
    Fee         Fee       `bson:"Fee"`
}

type CompanyBasicInfo struct {
    Id           primitive.ObjectID `bson:"_id,omitempty"`
    General      General             `bson:"General"`
}

File: api/models/company.js
const mongoose = require("mongoose");
const Schema = mongoose.Schema;

const PartnerSchema = {
  AadhaarName: String,
  DateOfBirth: Date,
  AadhaarNumber: String,
  Address: String,
  PAN: String,
  PANName: String,
  Mobile: String,
  Email: String,
  Photo: String,
};

const EPFOSchema = {
  EPFOStatus: String,
  EPFOLogin: String,
  EPFOPassword: String,
  EPFOCode: String,
  EPFOCodeType: String,
  EPFOMailId: String,
  EPFOMobile: String,
  EPFOFilingSince: Date,
  EPFODateOfCoverage: Date,
  EPFOCoverageType: String,
  DSCStatus: String,
  DSCExpiryDate: Date,
  DSCPassword: String,
  ESignStatusEPFO: String,
  EPFOOfficeName: String,
  EPFOOfficeAddress: String,
  EPFOOfficeContactNumber: String,
};

const ESICSchema = {
  ESICStatus: String,
  ESICLogin: String,
  ESICPassword: String,
  ESICCode: String,
  ESICCodeType: String,
  ESICMailId: String,
  ESICMobile: String,
  ESICFilingSince: Date,
  ESICDateOfCoverage: Date,
  ESICOfficeName: String,
  ESICOfficeAddress: String,
  ESICOfficeContactNumber: String,
};

const ShramSuvidhaSchema = {
  ShramSuvidhaLogin: String,
  ShramSuvidhaPassword: String,
  ShramSuvidhaMailId: String,
  ShramSuvidhaMobile: String,
  NearestPoliceStation: String,
  NICCode: String,
  BusinessPlace: String,
  DateOfApplication: Date,
  LeaseDateFrom: Date,
  LeaseDateTo: Date,
};

const IFSchema = {
  IFStatus: String,
  IFLogin: String,
  IFPassword: String,
  IFLicenseNumber: String,
  IFMailId: String,
  IFMobile: String,
  IFDateOfRegistration: Date,
  IFRenewalCharges: Number,
  IFRenwalSince: Date,
  NumberOfPressureVessels: Number,
  ISMLogin: String,
  ISMPassword: String,
};

const ContactSchema = {
  ContactName: String,
  Designation: String,
  Email: String,
  Mobile: String,
};

const FeeSchema = {
  Services: [String],
  Period: String,
  Fee: Number,
};

const GeneralSchema = {
  CompanyName: String,
  CompanyShort: String,
  CompanyId: Number,
  Services: [String],
  ESIPFCreatedBy: String,
  CompanyType: String,
  Status: String,
  BusinessAddress: String,
  City: String,
  PostalCode: String,
  District: String,
  State: String,
  ContactPersonName: String,
  ContactPersonRole: String,
  ContactPersonMobile: String,
  ContactPersonWhatsapp: String,
  ContactPersonMailId: String,
};

const CompanyDetailsSchema = {
  CompanyPAN: String,
  PANDate: Date,
  PANName: String,
  GSTDate: Date,
  BankAccountName: String,
  BankAccountNumber: String,
  IFSC: String,
  BankType: String,
  BankName: String,
  BranchName: String,
  LIN: String,
  CIN: String,
  CINDate: Date,
  BranchAddress: String,
  NatureOfBusiness: String,
  GSTNumber: String,
  City: String,
  PostalCode: String,
  District: String,
  State: String,
  PartnerList: [PartnerSchema],
};

const RegistrationSchema = {
  EPFO: EPFOSchema,
  ESIC: ESICSchema,
  ShramSuvidha: ShramSuvidhaSchema,
  IF: IFSchema,
};

const ContactListSchema = {
  ContactList: [ContactSchema],
};

const FeeModelSchema = {
  Fee: FeeSchema,
};

const CompanySchema = new Schema({
  General: GeneralSchema,
  CompanyDetails: CompanyDetailsSchema,
  Registration: RegistrationSchema,
  ContactList: ContactListSchema,
  Fee: FeeModelSchema,
});

const Company = mongoose.model("companies", CompanySchema);

module.exports = Company;
